src/pages/ProductListing/Filters/Filters.jsx

```diff
@@ -1,64 +1,127 @@
+import { CategoryCard } from "../../../components";
+import { useFilter, useProduct } from "../../../contexts";
 import "./Filters.css";
 export const Filters = () => {
+	const {
+		category,
+		rating,
+		sortByPrice,
+		priceRange,
+		filterDispatch: dispatch,
+		initialFilterState,
+	} = useFilter();
+
 	return (
 		<aside className="aside aside--products">
-			<form action="post" className="filter">
+			<form className="filter">
 				<header className="filter__header">
 					<h3 className="h5 filter__head">Filters</h3>
-					<input type="reset" value="Clear All" className="filter--reset" />
+					<button
+						onClick={(e) => {
+							e.preventDefault();
+							dispatch({ type: "RESET", payload: initialFilterState });
+						}}
+						className="filter--reset"
+					>
+						Clear All
+					</button>
 				</header>
 				<div className="filter">
 					<section className="filter__section">
 						<h3 className="h5 filter__head">Price</h3>
 
 						<ul className="range__head">
-							<li className="range__amount">500</li>
-							<li className="range__amount">7000</li>
+							<li className="range__amount">0</li>
+							<li className="range__amount">7500</li>
 							<li className="range__amount">15000</li>
 						</ul>
 
 						<input
+							onChange={(e) =>
+								dispatch({
+									type: "PRICE_RANGE",
+									payload: Number(e.target.value),
+								})
+							}
+							step="1000"
 							type="range"
-							min="500"
-							max="2500"
-							defaultValue="1500"
+							min="0"
+							max="15000"
+							value={priceRange}
 							className="range__sliders"
 						/>
+						{priceRange}
 					</section>
 					<section className="filter__section">
 						<h3 className="h5 filter__head">Category</h3>
 
 						<ul className="filter__items">
 							<li className="filter__list">
-								<input type="checkbox" id="Bat" className="filter__checkbox" />
-								<label htmlFor="Bat" className="filter__item_label">
+								<input
+									checked={category.includes("bats")}
+									onChange={() =>
+										dispatch({ type: "CATEGORY", payload: "bats" })
+									}
+									type="checkbox"
+									id="bats"
+									className="filter__checkbox"
+								/>
+								<label htmlFor="bats" className="filter__item_label">
```

Comment:
You can use here Input component may be that would be a great way to do it. And you can also reduce the line of code.

Reply:
Will look into it.